tutorials/llm/slimpajama/data_pipeline.ipynb

```diff
@@ -0,0 +1,328 @@
+    "3. Concatenating data\n",
+    "4. Preprocessing data for NeMo 2.0/Megatron\n",
+    "\n",
+    "First, we'll define each step. Next, we will see how we can use NeMo-Run to execute the steps sequentially on your local workstation using Docker or on Slurm.\n",
+    "\n",
+    "### Pre-requisites\n",
+    "This notebook assumes familiarity with [NeMo-Run](https://github.com/NVIDIA/NeMo-Run). Additionally, the docker execution and slurm execution steps require access to docker on your host and a remote slurm cluster respectively.\n",
+    "Additionally, you will have to complete the following steps:\n",
+    "\n",
+    "1. Set HOST_DATA_PATH in the first cell to a parent folder on your workstation where you want to save the data.\n",
+    "1. Create directories `HOST_DATA_PATH/tokenizer` and `HOST_DATA_PATH/slimpajama`.\n",
+    "1. Download the Llama 3 `tokenizer.model` file either from [Huggingface](https://huggingface.co/meta-llama/Llama-2-7b/blob/main/tokenizer.model) or https://www.llama.com/llama-downloads/ and place it at `{HOST_DATA_PATH}/tokenizer/tokenizer.model`.\n",
+    "    For HF, you can do it by running \n",
+    "    ```bash\n",
+    "    HF_TOKEN=... huggingface-cli download meta-llama/Llama-2-7B tokenizer.model --local-dir {HOST_DATA_PATH}/tokenizer/\n",
+    "    ```\n",
+    "\n",
+    "> [!NOTE]\n",
+    "> All code for this tutorial can be found at https://github.com/NVIDIA/NeMo/tree/main/examples/llm/slimpajama."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "\n",
+    "from data.download import download_slimpajama\n",
+    "from data.extract import run_extraction\n",
+    "from data.preprocess import preprocess_data\n",
+    "\n",
+    "HOST_DATA_PATH = \"/data\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Downloading Data\n",
+    "\n",
+    "First, we will configure the task to download data from Huggingface. We will use the Huggingface CLI for this. The function that configures the download script can be found [here](./data/download.py)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "download_task = download_slimpajama(\n",
+    "    include_pattern='--include \"train/chunk1/*_100*zst\"',\n",
+    ")\n",
+    "\n",
+    "# The configured script looks like below\n",
+    "print(download_task.inline)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extracting Data\n",
+    "\n",
+    "The downloaded data is in compressed zst format. We need to extract it into jsonl files. For that, we will configure the `extract_data` function defined [here](./data/extract.py). This function also allows excluding certain sources. By default, we exclude all data from `RedPajamaBook` set, but this setting is configurable."
```

Comment:
fix capitalization

The downloaded data is in compressed ZST format. We need to extract it into JSONL files. For that, we will configure the extract_data function defined here. This function also allows excluding certain sources. By default, we exclude all data from the RedPajamaBook set, but this setting is configurable.